src/tests/detect-ssh-hassh-server-string.c

```diff
@@ -0,0 +1,187 @@
+/* Copyright (C) 2019 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Vadym Malakhatko <malakhatkovadim@gmail.com>
+ *
+ */
+
+
+/**
+ * \test Test matching on a simple client hello packet
+ */
+static int DetectSshHasshServerStringTest01(void) {
```

Comment:
I think I'd prefer these tests to be done as pcap based suricata-verify tests.

See also https://redmine.openinfosecfoundation.org/issues/3725